huxley/www/static/js/huxley/components/CountrySelect.js

```diff
@@ -11,13 +11,20 @@ var CountrySelect = React.createClass({
   propTypes: {
     onChange: React.PropTypes.func,
     countries: React.PropTypes.array,
-    selectedCountryID: React.PropTypes.number
+    selectedCountryID: React.PropTypes.number,
+    countryPreferences: React.PropTypes.array
   },
 
   shouldComponentUpdate: function(nextProps, nextState) {
+    var preferenceChange = false;
+    for (var i = 0; i < this.props.countryPreferences.length; i++) {
+      if (this.props.countryPreferences[i] != nextProps.countryPreferences[i]) 
+        preferenceChange = true;
```

Comment:
If we know that the preferences have changed, we can return early, avoiding the `preferenceChange` variable, and falling back to that `return (... || ...)` statement otherwise.

```js
for (var i = 0; i < this.props.countryPreferences.length; i++) {
  if (this.props.countryPreferences[i] != nextProps.countryPreferences[i]) {
    return true;
  }
}
return (
  ... ||
);
```

Couple of style nits:
- prefer `===` and `!==` in JS, unless you specifically want type coercion. You can look up a famous talk on JS called "WAT" to see why.
- Always put braces after `if/for/while/etc` statements. Look up the Apple `goto fail` bug to see why.